test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs

```diff
@@ -26,25 +28,31 @@ public class SwaggerIntegrationTests
         [InlineData(typeof(OAuth2Integration.Startup), "/resource-server/swagger/v1/swagger.json")]
         [InlineData(typeof(ReDocApp.Startup), "/swagger/v1/swagger.json")]
         [InlineData(typeof(TestFirst.Startup), "/swagger/v1-generated/openapi.json")]
-        public async Task SwaggerEndpoint_ReturnsValidSwaggerJson(
+        public void SwaggerEndpoint_ReturnsValidSwaggerJson(
             Type startupType,
             string swaggerRequestUri)
         {
-            System.Net.Http.HttpClient client;
             var testSite = new TestSite(startupType);
 
             // Make sure not to build the same client in parallel
-            lock(startupType)
+            lock (startupType)
             {
-                client = testSite.BuildClient();
+                using (var client = testSite.BuildClient())
+                {
+                    var swaggerResponse = GetSwaggerResponse(client);
+
+                    swaggerResponse.EnsureSuccessStatusCode();
+                    var contentStream = swaggerResponse.Content.ReadAsStream();
+                    new OpenApiStreamReader().Read(contentStream, out OpenApiDiagnostic diagnostic);
+                    Assert.Empty(diagnostic.Errors);
+                }
             }
 
-            var swaggerResponse = await client.GetAsync(swaggerRequestUri);
+            HttpResponseMessage GetSwaggerResponse(HttpClient client)
+            {
+                return client.GetAsync(swaggerRequestUri).Result;
```

Comment:
This could potentially create deadlocks.

Making this class implement [IAsyncLifetime](https://www.danclarke.com/cleaner-tests-with-iasynclifetime) with a class fixture might be a way to create a serialized way to do this work ahead of the individual tests running.